src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/UserTimeZoneService.cs

```diff
@@ -54,25 +55,23 @@ public async ValueTask UpdateUserTimeZoneAsync(IUser user)
         return;
     }
 
-    public async ValueTask<string> GetCurrentUserTimeZoneIdAsync()
+    /// <inheritdoc/>
+    private async ValueTask<string> GetTimeZoneIdAsync(IUser user)
     {
-        var userName = _httpContextAccessor.HttpContext?.User?.Identity?.Name;
-
+        var userName = user.UserName;
         if (string.IsNullOrEmpty(userName))
         {
             return null;
         }
 
         var key = GetCacheKey(userName);
+
         var timeZoneId = await _distributedCache.GetStringAsync(key);
 
         // The timezone is not cached yet, resolve it and store the value
         if (string.IsNullOrEmpty(timeZoneId))
         {
-            // Delay-loading UserManager since it is registered as scoped
-            var userManager = _httpContextAccessor.HttpContext.RequestServices.GetRequiredService<UserManager<IUser>>();
-            var user = await userManager.FindByNameAsync(userName) as User;
-            timeZoneId = user.As<UserTimeZone>()?.TimeZoneId;
+            timeZoneId = (user as User).As<UserTimeZone>()?.TimeZoneId;
```

Comment:
I would using is instead of as to ensure it's not null.

Reply:
FYI I'm using the old code with my changes

Reply:
Okay but old code isn't correct. So I suggest fixing it.